src/reflect/scala/reflect/internal/Variances.scala

```diff
@@ -138,6 +137,9 @@ trait Variances {
         // than the result of the pattern match above, which normalizes types.
         tp
       }
+      private def shouldDealias(sym: Symbol): Boolean = {
+        sym.isAliasType && isLocalOnly(sym)
```

Comment:
Justify why we need to dealias for local only symbols:

```
trait C[-I] {
  private[this] type X = C[I]

  def f2(b: X): Unit
}

```

Because we didn't check `C[I]` in the RHS of the local only type alias, we need to expand the alias so we check it in the public signature of `f2`.